bundles/com.zeligsoft.domain.dds4ccm.ui/src/com/zeligsoft/domain/dds4ccm/ui/queries/HideStereotypeName.java

```diff
@@ -24,8 +24,12 @@
 
 	public String evaluate(NamedElement source, IParameterValueList2 parameterValues, IFacetManager facetManager)
 			throws DerivedTypedElementException {
-		if (ZDLUtil.isZDLProfile(source, DDS4CCMDomainFilter.DDS4CCM_PROFILE_NAME)) {
-			return labelProvider.getText(source);
+		try {
+			if (ZDLUtil.isZDLProfile(source, DDS4CCMDomainFilter.DDS4CCM_PROFILE_NAME)) {
+				return labelProvider.getText(source);
+			}
+		} catch (Exception e) {
```

Comment:
Catching `Exception` is almost certainly too broad, neither method has a `throws` clause. Catching `RuntimeException` is only slightly better. At least it communicates we're trying to suppress runtime exceptions. It is, however, extremely broad. If you are running 'untrusted code', but need to return, the making use of `SafeRunner` and an `ISafeRunnableWithException` might be a good alternative:

```
		final String result = SafeRunner.run(new ISafeRunnableWithResult<String>() {
			@Override
			public String runWithResult() throws Exception {
				if (ZDLUtil.isZDLProfile(source, DDS4CCMDomainFilter.DDS4CCM_PROFILE_NAME)) {
					return labelProvider.getText(source);
				}
				return null;
			}
		});
		return result != null ? result : defaultLabelProvider.getText(source);
```